src/System.Linq.Expressions/src/System/Linq/Expressions/NewExpression.cs

```diff
@@ -183,6 +183,8 @@ public static NewExpression New(ConstructorInfo constructor, IEnumerable<Express
         public static NewExpression New(ConstructorInfo constructor, IEnumerable<Expression> arguments, IEnumerable<MemberInfo> members)
         {
             ContractUtils.RequiresNotNull(constructor, nameof(constructor));
+            ContractUtils.RequiresNotNull(constructor.DeclaringType, nameof(constructor) + "." + nameof(constructor.DeclaringType));
```

Comment:
I added this to be consistent with the other simpler overload which checks for a non-null declaring type. 

This actually led me to post a stack overflow question. It is possible that DeclaringType is null, if the constructor is a module initialiser, supported by the CLR but not by c#. I've tried writing tests for this: the APIs are exposed in .NET fx but not .NET core. See: http://stackoverflow.com/questions/39240524/can-a-constructorinfo-have-no-declaringtype